CRM/Contact/Page/ImageFile.php

```diff
@@ -55,7 +55,8 @@ public function run() {
       CRM_Utils_System::civiExit();
     }
     else {
-      throw new CRM_Core_Exception(ts('Photo does not exist'));
+      header("HTTP/1.0 404 Not Found");
+      return;
```

Comment:
I would say rather than doing a return here we should call `CRM_Utils_System::civiExit()` but this could also be a follow up to this PR.

Reply:
Why would you suggest this a follow up? Or would more returns in civicrm code be candidates for CRM_Utils_System::civiExit() ?

Reply:
He might have just been politely trying to keep the PR from going on too long but if you want to update it now then you could pull the exit call out of the `if`, which funnily enough, makes it look more like it did [7 years ago](https://github.com/civicrm/civicrm-core/blob/94fbf0d091dbd7b5a36fb5d88683159787780fec/CRM/Contact/Page/ImageFile.php#L44-L57).

```
    if ($cid) {
      $config = CRM_Core_Config::singleton();
      $fileExtension = strtolower(pathinfo($_GET['photo'], PATHINFO_EXTENSION));
      $this->download(
        $config->customFileUploadDir . $_GET['photo'],
        'image/' . ($fileExtension == 'jpg' ? 'jpeg' : $fileExtension),
        $this->ttl
      );
    }
    else {
      header("HTTP/1.0 404 Not Found");
    }
    CRM_Utils_System::civiExit();
```

Reply:
I pulled the exit outside of the if now. @demeritcowboy @seamuslee001 agree & approve?

Reply:
Can you squash the commits?

Reply:
yep, done

Reply:
Thanks. Looks good - have updated label.